src/components/form-input.tsx

```diff
@@ -0,0 +1,19 @@
+import { ComponentProps } from "react";
+import Input from "./common/input";
+
+type FormInputProps = ComponentProps<"input"> & {
+  error?: string;
+};
+
+export default function FormInput({ error, ...rest }: FormInputProps) {
+  return (
+    <>
+      <Input {...rest} />
+      {error && (
+        <span className="ml-[5px] mt-[5px] text-[13px] leading-[18px] text-red-500">
+          {error}
+        </span>
+      )}
+    </>
+  );
+}
```

Comment:
```suggestion
const FormInput = forwardRef<HTMLInputElement, FormInputProps>(
  ({ error, ...rest }, ref) => {
    return (
      <>
        <Input ref={ref} {...rest} />
        {error && (
          <span className="ml-[5px] mt-[5px] text-[13px] leading-[18px] text-red-500">
            {error}
          </span>
        )}
      </>
    );
  },
);
FormInput.displayName = "FormInput";
```
React hook form sẽ dùng ref để quản lí state.
em phải forwardRef hoặc 
` {...register("email")}
inputRef = register("email").ref`
sau đó trong formInput 
ref={rest.inputRef}

Nhưng khuyến khích dùng forward ref